### PR TITLE
Update live from master

### PR DESCRIPTION
**Pull request by hayfield**
Adds tests and some things around DAC sector codes for comprehensiveness.

### CODE REVIEW

File: stats/dashboard.py [hunk 1]
```diff
@@ -1003,41 +1003,52 @@ def is_text_in_element(elementName):
...
-                all_and_not_empty(self.element.xpath('default-aid-type/@code'))
-                or all_and_not_empty([transaction.xpath('aid-type/@code') for transaction in self.element.xpath('transaction')])
+                all_true_and_not_empty(self.element.xpath('default-aid-type/@code'))
+                or all_true_and_not_empty([transaction.xpath('aid-type/@code') for transaction in self.element.xpath('transaction')])
                 )
-            # Alternative: all(map(all_and_not_empty, [transaction.xpath('aid-type/@code') for transaction in self.element.xpath('transaction')]))
+            # Alternative: all(map(all_true_and_not_empty, [transaction.xpath('aid-type/@code') for transaction in self.element.xpath('transaction')]))
         }
 
+    def _is_sector_dac(self):
```
Comment (dalepotter) [inline stats/dashboard.py]: Needs a docstring - even if the rest of the code doesn't have docstrings, let's adhere to the [boy scout rule](http://deviq.com/boy-scout-rule/) when we interact with it.

File: stats/dashboard.py [hunk 2]
```diff
@@ -41,8 +41,8 @@ def add_years(d, years):
         return d + (date(d.year + years, 1, 1) - date(d.year, 1, 1))
 
 
-def all_and_not_empty(bool_iterable):
-    """ For a given list, check that all elements return true and that the list is not empty """
+def all_true_and_not_empty(bool_iterable):
+    """ For a given list, check that all elements return true and that the list is not empty."""
```
Comment (dalepotter) [inline stats/dashboard.py]: `bool_iterable` needs documenting :-)

File: stats/tests/test_comprehensiveness.py [hunk 3]
```diff
@@ -46,11 +46,11 @@ def test_comprehensiveness_is_current(major_version):
 
     def end_planned_date(datestring):
         """
-        Create an activity_stats element with a specified 'end-planned' date.  
+        Create an activity_stats element with a specified 'end-planned' date.
         Also sets the current date to 9990-06-01
 
         Keyword arguments:
-        datestring -- An ISO date to be used as the 'end-planned' date for the 
+        datestring -- An ISO date to be used as the 'end-planned' date for the
```
Comment (dalepotter) [inline stats/tests/test_comprehensiveness.py]: datestring {str) ??

File: statsrunner/loop.py [hunk 4]
```diff
@@ -4,16 +4,23 @@
...
 import statsrunner.shared
 import statsrunner.aggregate
 from statsrunner.common import decimal_default
 
+
 def call_stats(this_stats, args):
+    """Create dictionary of enabled stats for this_stats object.
+
+    Args:
+      this_stats (cls): stats_module that specifies calculations for relevant input, processed by internal methods of process_file().
```
Comment (dalepotter) [inline statsrunner/loop.py]: Presumably `cls` should be `stats.dashboard.CommonSharedElements`, but I could be wrong ??

Comment (dalepotter) [reply statsrunner/loop.py]: Also `args` should be listed as an... arg!

File: statsrunner/loop.py [hunk 5]
```diff
@@ -68,47 +88,67 @@ def process_stats_element(ElementStats, tagname=None):
...
             traceback.print_exc(file=sys.stdout)
             continue
     return files
 
+
 def loop(args):
+    """Loops through all specified folders to convert data to JSON output.
+
+    Args:
+        args: Object containing program run options (set by CLI arguments at runtime. See __init__ for more details).
```
Comment (dalepotter) [inline statsrunner/loop.py]: `args` -> `args (argparse.ArgumentParser)`

Comment (hayfield) [reply statsrunner/loop.py]: The `parser` is one of them, `args` is the output of that.

**Pull request merged**